package com.hejazi.distributed_payment_gateway.merchant.service;

import com.hejazi.distributed_payment_gateway.merchant.dto.request.UpdateWebhookConfigRequest;
import com.hejazi.distributed_payment_gateway.merchant.dto.response.WebhookConfigResponse;

import java.util.List;
import java.util.UUID;

public interface WebhookConfigService {
    WebhookConfigResponse create(UUID merchantId, UpdateWebhookConfigRequest request);

    List<WebhookConfigResponse> list(UUID merchantId);

    WebhookConfigResponse getById(UUID merchantId, UUID configId);

    WebhookConfigResponse update(UUID merchantId, UUID configId, UpdateWebhookConfigRequest request);

    void delete(UUID merchantId, UUID configId);
}
